client/environment_test.go

```diff
@@ -212,6 +222,51 @@ var _ = Describe("Environment Client", func() {
 		})
 	})
 
+	Describe("EnvironmentCreateWithoutTemplate", func() {
+		var createdEnvironment Environment
+		var err error
+
+		BeforeEach(func() {
+			mockOrganizationIdCall(organizationId)
+			createEnvironmentPayload := EnvironmentCreate{}
+			copier.Copy(&createEnvironmentPayload, &mockEnvironment)
+			createTemplatePayload := TemplateCreatePayload{}
+			copier.Copy(&createTemplatePayload, &mockTemplate)
+
+			createRequest := EnvironmentCreateWithoutTemplate{
+				EnvironmentCreate: createEnvironmentPayload,
+				TemplateCreate:    createTemplatePayload,
+			}
+
+			expectedCreateRequest := createRequest
+			expectedCreateRequest.TemplateCreate.OrganizationId = organizationId
+
+			httpCall = mockHttpClient.EXPECT().
+				Post("/environments/without-template", expectedCreateRequest, gomock.Any()).
+				Do(func(path string, request interface{}, response *Environment) {
+					*response = mockEnvironment
+				})
+
+			createdEnvironment, err = apiClient.EnvironmentCreateWithoutTemplate(createRequest)
+		})
+
+		It("Should get organization id", func() {
+			organizationIdCall.Times(1)
+		})
+
+		It("Should send POST request", func() {
+			httpCall.Times(1)
+		})
+
+		It("Should not return error", func() {
+			Expect(err).To(BeNil())
+		})
+
+		It("Should return the created environment", func() {
+			Expect(createdEnvironment).To(Equal(mockEnvironment))
+		})
```

Comment:
I think we are missing a was template creation called assertion here

Reply:
@ItamarMalka - we don't call the create template.
The API call creates both environment and template in the same API call.